src/System.Diagnostics.Contracts/tests/BasicContractsTests.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Contracts;
+using Xunit;
+
+namespace System.Diagnostics.Contracts.Tests
+{
+    public class BasicContractsTest
+    {
+        [Fact]
+        public static void AssertTrueDoesNotRaiseEvent()
+        {
+            bool eventRaised = false;
+            Contract.ContractFailed += (s, e) =>
+            {
+                eventRaised = true;
+                e.SetHandled();
+            };
+
+            Contract.Assert(true);
+            Assert.False(eventRaised, "ContractFailed event was raised");
+        }
+
+        [Fact]
+        public static void AssertFalseRaisesEvent()
+        {
+            bool eventRaised = false;
+            Contract.ContractFailed += (s, e) =>
+            {
+                eventRaised = true;
+                e.SetHandled();
+            };
+
+            Contract.Assert(false);
```

Comment:
What happens when these tests are compiled in release rather than debug, given that Assert is Conditional("DEBUG")?